core/src/client/client_api.rs

```diff
@@ -0,0 +1,118 @@
+use eyre::Result;
+
+// TODO: all reth_primitives::rpc types should me replaced when native reth Log is implemented
+// https://github.com/paradigmxyz/reth/issues/1396#issuecomment-1440890689
+use reth_primitives::{Address, BlockNumberOrTag, Bytes, H256, U256, U64};
+use reth_rpc_types::{
+    BlockTransactions, RichBlock, SyncStatus, Transaction as EtherTransaction, TransactionReceipt,
+};
+use starknet::{
+    core::types::FieldElement,
+    providers::jsonrpc::{
+        models::{BlockId as StarknetBlockId, Transaction as StarknetTransaction},
+        JsonRpcClientError,
+    },
+};
+
+use thiserror::Error;
+extern crate hex;
+use crate::helpers::MaybePendingStarknetBlock;
+
+use crate::client::types::TokenBalances;
+use async_trait::async_trait;
+use reth_rpc_types::Index;
+
+#[derive(Debug, Error)]
+pub enum KakarotClientError {
+    #[error(transparent)]
+    RequestError(#[from] JsonRpcClientError<reqwest::Error>),
+    #[error(transparent)]
+    OtherError(#[from] anyhow::Error),
+}
+
+#[async_trait]
+pub trait KakarotClient: Send + Sync {
+    async fn block_number(&self) -> Result<U64, KakarotClientError>;
+
+    async fn get_eth_block_from_starknet_block(
+        &self,
+        block_id: StarknetBlockId,
+        hydrated_tx: bool,
+    ) -> Result<RichBlock, KakarotClientError>;
+
+    async fn get_code(
+        &self,
+        ethereum_address: Address,
+        starknet_block_id: StarknetBlockId,
+    ) -> Result<Bytes, KakarotClientError>;
+
+    async fn call_view(
+        &self,
```

Comment:
since we're here, isn't there a way to have cargo fmt enforce a systematic spacing between blocks? here's one line break, below no

Reply:
✅